src/PostgREST/Query.hs

```diff
@@ -298,7 +298,7 @@ setPgLocals dbActPlan AppConfig{..} claims role ApiRequest{..} = lift $
     roleSettingsSql = setConfigWithDynamicName <$> HM.toList (fromMaybe mempty $ HM.lookup role configRoleSettings)
     appSettingsSql = setConfigWithDynamicName <$> (join bimap toUtf8 <$> configAppSettings)
     timezoneSql = maybe mempty (\(PreferTimezone tz) -> [setConfigWithConstantName ("timezone", tz)]) $ preferTimezone iPreferences
-    funcSettingsSql = setConfigWithDynamicName <$> (join bimap toUtf8 <$> funcSettings)
+    funcSettingsSql = setConfigWithDynamicName <$> (join bimap toUtf8 <$> filter (\(a,_) -> a `elem` configDbHoistedTxSettings) funcSettings)
```

Comment:
@wolfgangwalther By the changing the filter, io-tests are failing as the config is not applying now.
```sql
LEFT JOIN LATERAL (
    SELECT
      array_agg(row(
        substr(setting, 1, strpos(setting, '=') - 1),
        lower(substr(setting, strpos(setting, '=') + 1))
      )) as kvs
    FROM unnest(proconfig) setting
    WHERE setting NOT LIKE ALL(ARRAY['|] <> encodeUtf8 (T.intercalate "," hoistedSets) <> [q|'])
  ) func_settings ON TRUE
  WHERE t.oid <> 'trigger'::regtype AND COALESCE(a.callable, true)
|] <> (if pgVer >= pgVersion110 then "AND prokind = 'f'" else "AND NOT (proisagg OR proiswindow)")
```
I get,
```
FAILED test/io/test_io.py::test_first_hoisted_setting_is_applied - assert '{"get_work_m...imeout":"7s"}' == '{"get_work_m...imeout":"2s"}'
FAILED test/io/test_io.py::test_second_hoisted_setting_is_applied - assert '{"get_work_m...imeout":"7s"}' == '{"get_work_m...imeout":"7s"}'
============= 2 failed, 157 passed, 1 skipped in 67.99s (0:01:07) ==============

```

Reply:
It's not applying because:

```sql
    WHERE setting NOT LIKE ALL(ARRAY['|] <> encodeUtf8 (T.intercalate "," hoistedSets) <> [q|'])
```

Will return:

```sql
    WHERE setting NOT LIKE ALL(ARRAY['statement_timeout,plan_filter.statement_cost_limit,default_transaction_isolation'])
```

That is, all the settings as a single element of the array.

Also, since this is a dynamic value, I think this should be parameterized. You can do something similar to what it's done here:

https://github.com/PostgREST/postgrest/blob/11c9e8dac903aa27511defa390383bcecd256681/src/PostgREST/SchemaCache.hs#L937-L939
https://github.com/PostgREST/postgrest/blob/11c9e8dac903aa27511defa390383bcecd256681/src/PostgREST/SchemaCache.hs#L980